From 97791e7c7cdab071dbaad408969e59bd99ef2bcf Mon Sep 17 00:00:00 2001
From: "Christopher N. Hesse" <raymanfx@gmail.com>
Date: Wed, 4 Apr 2018 12:15:13 +0200
Subject: [PATCH 1/2] recovery/ui: Hide emulated storage for encrypted devices

We do not support decryption of data partitions, so get
rid of the option in the UI.

Change-Id: Idcccba84ea0ba42c24f8a52b7f0994ed4545475d
---
 recovery.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/recovery.cpp b/recovery.cpp
index ecc2df7f..995da12f 100644
--- a/recovery.cpp
+++ b/recovery.cpp
@@ -1177,6 +1177,11 @@ refresh:
     items.push_back(MenuItem("Apply from ADB")); // Index 0
 
     for (auto& vol : volumes) {
+        // On devices where /data is encrypted (FDE or FBE), we cannot install from
+        // internal (aka emulated) storage because we do not support decryption.
+        if (ensure_path_mounted(vol.mPath.c_str()) != 0) {
+            continue;
+        }
         items.push_back(MenuItem("Choose from " + vol.mLabel));
     }
 
-- 
2.17.1

